arcs_fact_tests/fail_different_tag.dl

```diff
@@ -0,0 +1,31 @@
+// Original Arcs manifest:
+// Fails when different tag is claimed.
+// #Ingress: P1
+// #Fail: hc:P2.bar is trusted
+//particle P1
+//  foo: writes Foo {}
+//  claim foo is notTrusted
+//particle P2
+//  bar: reads Foo {}
+//  check bar is trusted
+//recipe R
+//  P1
+//    foo: writes h
+//  P2
+//    bar: reads h
+#include "taint.dl"
+#include "analyze_tag_checks_test_helper.dl"
+
+claimHasTag("R.P1.foo", "notTrusted").
+edge("R.P1.foo", "R.h.Foo").
+edge("R.h.Foo", "R.P2.bar").
+
+// This check was in the original manifest. We expect it not to pass.
+checkTag("R.P2.bar", "trusted", "PRESENT").
+
+// This check was not in the original manifest, but we add it to make sure the dataflow did
+// something.
+checkTagExtra("R.P2.bar", "notTrusted", "PRESENT").
+
+tagCheckExpectation("R.P2.bar", "trusted", "FAIL").
+tagCheckExpectation("R.P2.bar", "notTrusted", "PASS").
```

Comment:
What do these statements mean?

Reply:
These are setting up our expectations for how the checks expressed will evaluate. You can read them as follows:

`tagCheckExpectation("R.P2.bar", "trusted", "FAIL").` -> We expect the check that asserts the (presence|absence) of the tag `trusted` on `R.P2.bar` to fail.
`tagCheckExpectation("R.P2.bar", "notTrusted", "PASS").` -> We expect the check that asserts the (presence|absence) of the tag `notTrusted` on `R.P2.bar` to pass.

The original manifest checks that `R.P2.bar` has the `trusted` tag. However, inspecting the dataflow shows that this should not be the case; no `trusted` tag is claimed anywhere in the manifest. Therefore, if the dataflow analysis is behaving properly, we expect this check to fail, giving us the first `tagCheckExpectation` fact.

Now there are two ways that the above expectation could be fulfilled: our dataflow could be working as intended, or it could just be completely broken and not propagating tags at all. To check that it's doing something, I added an "extra" check not present in the original manifest checking that `notTrusted` is propagated to `R.P2.bar`. I then added the additional expectation that this check passes, producing the second `tagCheckExpectation` fact.

Reply:
Removed as per your recommendation.